src/include/detail/graph/adj_list.h

```diff
@@ -143,7 +143,7 @@ class adj_list : public std::vector<std::list<std::tuple<SC, ID>>> {
   adj_list(AdjList&& l)
       : Base(l.size()) {
     for (size_t i = 0; i < l.size(); ++i) {
-      for (auto&& [dst, val] : l[i]) {
+      for (auto&& [val, dst] : l[i]) {
         add_edge(i, dst, val);
       }
     }
```

Comment:
@lums658 how was our code working before this? Should unit tests have been failing? If not, are there unit tests we can write that will fail if we change this back to the incorrect version?

Reply:
Unit test added.